File: src/main/java/mk/ukim/finki/gamealytical/service/GamealyticalService.java
package mk.ukim.finki.gamealytical.service;

import java.util.List;
import java.util.Map;

public interface GamealyticalService {

    List<Map<String, String>> getAllVideoGames();

    Map<String, String> getVideoGameByName(String name);

    List<Map<String, String>> getFeaturedVideoGames();

}

File: gamealytical-frontend/src/Services/GamealyticalService.js
import AxiosGamealytical from "../Utilities/AxiosHttpClient/AxiosHttpClient";

const GamealyticalService = {
    getAllVideoGames: () => {
        return AxiosGamealytical.get("/api/gamealytical");
    },
    getVideoGameByName: (name) => {
        return AxiosGamealytical.get(`/api/gamealytical/find?name=${name}`);
    },
    getFeaturedVideoGames: () => {
        return AxiosGamealytical.get("/api/gamealytical/featured");
    }
}

export default GamealyticalService;